### PR TITLE
Feature/reset notif statistics

### PR DESCRIPTION
**Pull request by crbrox**
@fgalan 
@kzangeli

### CODE REVIEW

File: src/lib/logMsg/logMsg.h [hunk 1]
```diff
@@ -1812,11 +1812,11 @@ inline void lmTransactionSetService(const char* _service)
 {
   if (strlen(_service) != 0)
   {
-    snprintf(service, sizeof(service), _service);
+    snprintf(service, sizeof(service), "%s", _service);
```
Comment (kzangeli) [inline src/lib/logMsg/logMsg.h]: Funny that the compiler didn't complain about this ...

NTC

**Comment by kzangeli** [comment]
LGTM

File: src/lib/serviceRoutines/statisticsTreat.cpp [hunk 2]
```diff
@@ -245,14 +247,17 @@ std::string renderSemWaitStats(void)
...
+  float      timeInQ = QueueStatistics::getTimeInQ();
+  int        out     = QueueStatistics::getOut();
+
+  jh.addNumber("in",             QueueStatistics::getIn());
+  jh.addNumber("out",            out);
+  jh.addNumber("reject",         QueueStatistics::getReject());
+  jh.addNumber("sentOk",         QueueStatistics::getSentOK());     // FIXME P7: this needs to be generalized for all notificationModes
+  jh.addNumber("sentError",      QueueStatistics::getSentError());  // FIXME P7: this needs to be generalized for all notificationModes
+  jh.addFloat ("timeInQueue",    timeInQ);
+  jh.addFloat ("avgTimeInQueue", out==0 ? 0 : (timeInQ/out));
```
Comment (fgalan) [inline src/lib/serviceRoutines/statisticsTreat.cpp]: Is the timeInQ/out casting correctly to float? I have found problems with this kind of things sometimes with C/C++.

Just to check that you have actually test it and it works ;)

File: src/lib/rest/httpRequestSend.cpp [hunk 3]
```diff
@@ -471,7 +471,7 @@ int httpRequestSendWithCurl
 
   lmTransactionEnd();
 
-  return 0;
+  return res == CURLE_OK ? 0 : -9;
```
Comment (fgalan) [inline src/lib/rest/httpRequestSend.cpp]: Maybe this involves a fix that worth to be mentioned in CNR?

Comment (kzangeli) [inline src/lib/rest/httpRequestSend.cpp]: (I missed this detail)
Especially it should be mentioned in the function header with the other error codes

Comment (crbrox) [inline src/lib/rest/httpRequestSend.cpp]: Fixed 5ef5947d7cb1a0863f589c83cb23c2c95357a164

File: src/lib/serviceRoutines/statisticsTreat.cpp (hunk 2 above)
Comment (fgalan) [inline src/lib/serviceRoutines/statisticsTreat.cpp]: After skype discussion, NTC.

File: src/lib/rest/httpRequestSend.cpp (hunk 3 above)
Comment (crbrox) [inline src/lib/rest/httpRequestSend.cpp]: CNR fixed e1230e2

**Comment by fgalan** [comment]
LGTM

**Pull request merged**